tuning/sft_trainer.py

```diff
@@ -360,7 +361,24 @@ def train(
         for x in framework.get_callbacks_and_ready_for_train(model, accelerator):
             trainer.add_callback(x)
 
-    trainer.train()
+    resume_from_checkpoint = None
+    # Check if resume flag is not passed (None), or if flag is true and
+    # output_dir has checkpoints then get last checkpoint from output_dir
+    if (
+        training_args.resume_from_checkpoint is None
+        or training_args.resume_from_checkpoint.lower() == "true"
```

Comment:
Since this does accept a boolean value, shouldn't this just check the bool value? A user isn't passing in a string "True" but the boolean "True" right?

Reply:
Yes, `resume_from_checkpoint` does accepts bool in `trainer.train()`. Its just that, when user passes `resume_from_checkpoint` flag as bool, it gets taken as string as it gets treated as flag in [training arguments](https://huggingface.co/docs/transformers/main_classes/trainer#transformers.TrainingArguments.resume_from_checkpoint) which just accepts string.
<img width="898" alt="image" src="https://github.com/user-attachments/assets/af25630a-de41-4899-9afc-b245f3aa4705">

Reply:
ahh thanks for the clarity